kotlinpoet/src/test/java/com/squareup/kotlinpoet/FunSpecTest.kt

```diff
@@ -429,6 +429,20 @@ class FunSpecTest {
     )
   }
 
+  @Test fun functionWithContextReceiver() {
+    val stringType = STRING
+    val funSpec = FunSpec.builder("foo")
+      .contextReceiver(stringType)
+      .build()
+
+    assertThat(funSpec.toString()).isEqualTo(
+      """
+      |context(kotlin.String) public fun foo(): kotlin.Unit {
```

Comment:
The receiver should be the line above the function right?

Also please add more thorough tests for this:
- check that it fails on constructors
- add multiple types
- add generic types (can these be generic and reference generic types in the function descriptor?)
- add a typevariablename (or if that's not legal, add a check + test)
- typenames with annotations (or a check + test if that's not legal)